tests/macsec/test_neighbor_reboot.py

```diff
@@ -0,0 +1,47 @@
+'''
+
+The test case will verify the behavior when a neighbor reboots while Macsec is Active.
+
+Step 1: Configure Macsec between neighbor and DUT
+Step 2: Save config on neighbor and reboot
+Step 3: Verify macsec connection is re-established after reboot
+
+'''
+
+import pytest
+import logging
+from tests.common.utilities import wait_until
+from tests.common.helpers.assertions import pytest_assert
+
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.macsec_required,
+    pytest.mark.topology("t2")
+]
+
+
+def test_neighbor_reboot(duthosts, request, enum_frontend_dut_hostname, ctrl_links):
```

Comment:
Thanks for this test. But I think the idea here is already tested/covered in another test where we do a link flap by doing a remote interface shut
 https://github.com/sonic-net/sonic-mgmt/blob/fa5ab2566c9e30a6cbdaab87345eadd9ae63e7b2/tests/macsec/test_fault_handling.py#L41. 

I feel here we are testing the same scenario, instead of a remote interface shut -- here we do a neighbor VM reboot.

Reply:
@judyjoseph How would you like to move forward with this test case?  Should I remove the PR as a duplicate or implement the other feedback and continue?

Reply:
@judyjoseph , can we make a conclusion on this test plan?

Reply:
@judyjoseph How should we proceed?